analyses/mutation-signatures/util/mut_sig_functions.R

```diff
@@ -0,0 +1,209 @@
+  #
+  # Args:
+  #
+  #   which_sig_list: a list of `whichSignature` output for each sample
+  #   label: the label you would like associated with the plot as a character string.
+  #   sample_ids: vector of the sample ids.
+  #   output_dir: where the plots should be saved. If this directory doesn't exist,
+  #               this will create it. Default is current directory.
+  # Returns:
+  #   Saved png mutation signature plots to the specified directory.
+
+  if (!dir.exists(output_dir)) {
+    dir.create(output_dir, recursive = TRUE)
+  }
+
+  for (sample_num in 1:length(which_sig_list)) {
+    # Set up png
+    png(file.path(output_dir, paste0(sample_ids[sample_num], "_", label, "_mutation_sig.png")))
+    # Use the deconstructSigs function 
+    plotSignatures(which_sig_list[[sample_num]], sub = sample_ids[sample_num])
+    dev.off()
+  }
+}
+
+calc_mut_per_sig <- function(which_sig_list,
+                             muts_per_sample,
+                             wgs_genome_size,
+                             wxs_exome_size,
+                             maf_df) {
+  # Given a list of `deconstructSigs::whichSignature` output, calculate the
+  # mutations per signature per Mb of the genome/exome.
+  #
+  # Args:
+  #
+  #   which_sig_list: a list of `whichSignature` output
+  #   muts_per_sample: a vector with the total mutation signature counts for each 
+  #                    sample as calculated by deconstructSigs::mut.to.sigs.input 
+  #   wgs_genome_size: size of the WGS genome in bp
+  #   wxs_genome_size: size of the WXS exome in bp
+  #   maf_df: a data.frame with `short_histology` and `experimental strategy`  
+  #           information columns
+  #
+  # Returns:
+  #   A data.frame that is samples x signatures and has the number of mutations
+  #   per mb for each signature x sample combo
+
+  # Count the total number of signature mutations for each sample
+  total_muts <- apply(sigs_input, 1, sum)
+
+  # Pull out the signature weights and make into matrix
+  sig_weights <- do.call(
+    "rbind.data.frame",
+    lapply(which_sig_list, function(sample_data) sample_data$weights)
+  ) %>%
+    tibble::rownames_to_column("Tumor_Sample_Barcode")
+
+  # Calculate the number of mutations contributing to each signature
+  # Here the weight is multiplied by the total number of signature mutations. 
+  sig_num_df <- apply(sig_weights[, -1], 2, function(row_weights) row_weights * total_muts) %>%
+    as.data.frame() %>%
+
+    # Add back the sample ids
+    dplyr::mutate(Tumor_Sample_Barcode = sig_weights$Tumor_Sample_Barcode) %>%
+
+    # Join the short_histology and experimental stategy information
+    dplyr::inner_join(dplyr::select(
+      maf_df,
+      "Tumor_Sample_Barcode",
+      "short_histology",
+      "experimental_strategy"
+    ),
+    by = "Tumor_Sample_Barcode"
+    ) %>%
+    # Get rid of Panel samples
+    dplyr::filter(experimental_strategy != "Panel") %>%
+
+    # Reformat for plotting
+    reshape2::melt() %>%
```

Comment:
Give the variable column a good name?

Reply:
Yes! I ended up doing that but hadn't merged it yet!

Reply:
And you still haven't? Oh, you did below, but you can do it here:

```
reshape2::melt(varnames = c("Tumor_Sample_Barcode ", "signature"),
               value.name="num_mutations") 
```

either way...

Reply:
I like this better. Will do this!